package.json

```diff
@@ -17,9 +17,9 @@
     "exec-integ": "jest --config jestconfig.integ.json",
     "watch": "npm run test -- --watch",
     "build": "tsc",
-    "format": "prettier --write \"src/**/*.ts\" \"src/**/*.js\"",
-    "lint": "npm run fix-lint && tslint -p tsconfig.json",
```

Comment:
Absolutely!

So the primary reason for linting is to maintain a bar for quality for the codebase and to ensure that future commits at least meet that bar.

When you `fix-lint` during the normal linting process, this allows for code to get committed into the repo that doesn't meet that bar which happened recently in [this commit](https://github.com/awslabs/aws-embedded-metrics-node/commit/a63ea2fce7b29d7bff70392cc89301ac062b62ab).

The committer hadn't ran `lint` and, during the CodeBuild process, the issues they introduced were fixed during build time effectively swallowing the issues and allowing the code with linting issues to be committed into the repo.

Linting issues should fail the build during linting so that reviewers are able to see these errors and request second revisions.

With this change, that commit wouldn't have been able to be deployed since your AWS CodeBuild process (I assume, correct me if I'm wrong. If it isn't, it should be) would run the `lint` script and fail instead of swallowing the errors.

Reply:
Very good points and yes that's exactly what happened. The CI build fixed the lint errors but didn't commit the fixes (as it shouldn't). I agree that this is the right approach and thanks for taking the time to clarify.

https://github.com/awslabs/aws-embedded-metrics-node/blob/36b1951d2630befb7953d1566c2fca4756ac91cb/buildspec.yml#L21

Reply:
Will do! Thanks for the quick replies!